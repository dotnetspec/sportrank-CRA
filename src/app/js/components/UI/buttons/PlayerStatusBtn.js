import React from 'react';
import { Button } from 'react-bootstrap';
import JSONops from '../../Logic/JSONops';
/**
 * Functional component representing player status button in the header.
 * The component rendering in this area is controlled by
 * an @external "BrowserRouter"
 *
 * @extends React.Component
 */
 const PlayerStatusBtn = (username) => {

   const [playerActive, setPlayerActive] = React.useState(true);
   const [btnText, setValue] = React.useState('De-Activate?');
   const [bsStyle, setStyle] = React.useState('success');


   //const onChange = event => setValue(event.target.value);
     const _handleChangeStatusPlayerBtnText = (username, props) => {
     // const _handleChangeStatusPlayerBtnText = event =>
     // {
       if(username !== null){
           if(playerActive === true){
             setPlayerActive(false);
             setStyle('warning');
             setValue('Re-Activate?');
           }else{
             setPlayerActive(true );
             setStyle('success');
             setValue('De-Activate?')
             try {
               console.log('in _handleReactivatePlayer', props.newrankIdCB, this.props.rankingJSONdata, username, this.props.account)
               JSONops.reactivatePlayer(this.props.newrankIdCB, this.props.rankingJSONdata, username, this.props.account);
               this.props.history.push('/home/@' + username);
             } catch (err) {
             // stop loading state and show the error
             console.log(err.message);
             };
           }
         }else {
             console.log('no username passed to deactivate btn!')
           }
         }


   return (
       <Button className='deactivatebtn'  data-cy='deactivate'
       placeholder="De-Activate?"
       data-testid="activatebtn-input"
       bsStyle={bsStyle}
       onClick={(e) => _handleChangeStatusPlayerBtnText(username)}
       >
         {btnText}
       </Button>
   );
 };


 export default PlayerStatusBtn
