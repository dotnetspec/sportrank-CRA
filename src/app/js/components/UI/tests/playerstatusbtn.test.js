// src/__tests__/example.js
// query utilities:
import {
  // Tip: all queries are also exposed on an object
  // called "queries" which you could import here as well
  wait,
} from '@testing-library/dom'
// adds special assertions like toHaveTextContent
import 'jest-dom/extend-expect'
import PlayerStatusBtn from '../buttons/PlayerStatusBtn'
import React from 'react'
import { render, cleanup } from '@testing-library/react'

//originally based on example:
//https://testing-library.com/docs/dom-testing-library/example-intro

afterEach(cleanup);

fit('PlayerStatusBtn correctly displays initial status', async () => {
  //const playerStatusBtnText = 'Re-Activate?'

  const props  = {
    isCurrentUserActive:true
  }

  const historyMock = { push: jest.fn() };

  const { getByTestId} = render(<PlayerStatusBtn username='player1'
  history={historyMock}
  
  {...props}
  />)
  //const inputNode = getByPlaceholderText('De-Activate?')

  //inputNode.btnText = playerStatusBtnText

  expect(getByTestId('activatebtn-input')).toHaveTextContent(
    'De-Activate?'
  )

  // Get elements by their text, just like a real user does.
  //getByText(container, 'De-Activate?').click()
  //inputNode.click();

  // getByTestId and queryByTestId are an escape hatch to get elements
  // by a test id (could also attempt to get this element by it's text)
  // jest snapshots work great with regular DOM nodes!
  //expect(inputNode).toMatchSnapshot()
})

test('PlayerStatusBtn text on click', async () => {
  const playerStatusBtnText = 'Re-Activate?'

  const historyMock = { push: jest.fn() };

  const { getByPlaceholderText, getByTestId} = render(<PlayerStatusBtn username='player1'
  history={historyMock}
  />)
  const inputNode = getByPlaceholderText('De-Activate?')

  inputNode.btnText = playerStatusBtnText

  // Get elements by their text, just like a real user does.
  //getByText(container, 'De-Activate?').click()
  inputNode.click();

  await wait(() =>
    expect(getByTestId('activatebtn-input')).toHaveTextContent(
      playerStatusBtnText
    )
  )
  // getByTestId and queryByTestId are an escape hatch to get elements
  // by a test id (could also attempt to get this element by it's text)
  // jest snapshots work great with regular DOM nodes!
  expect(inputNode).toMatchSnapshot()
})
