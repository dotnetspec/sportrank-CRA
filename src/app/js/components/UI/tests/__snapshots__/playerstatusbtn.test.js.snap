// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerStatusBtn text correct with isCurrentUserActive = false 1`] = `
<button
  class="deactivatebtn btn btn-default"
  data-cy="deactivate"
  data-testid="activatebtn-input"
  placeholder="De-Activate?"
  style="color: green;"
  type="button"
>
  Re-Activate?
</button>
`;

exports[`PlayerStatusBtn text correct with isCurrentUserActive = true 1`] = `
<button
  class="deactivatebtn btn btn-default"
  data-cy="deactivate"
  data-testid="activatebtn-input"
  placeholder="De-Activate?"
  style="color: red;"
  type="button"
>
  De-Activate?
</button>
`;
